diff -Naur rpi-firmware.orig/rpi-firmware.mk rpi-firmware/rpi-firmware.mk
--- a/package/rpi-firmware/rpi-firmware.mk	2017-03-20 10:27:10.261649585 +0000
+++ b/package/rpi-firmware/rpi-firmware.mk	2017-03-20 10:27:36.760396542 +0000
@@ -4,7 +4,7 @@
 #
 ################################################################################
 
-RPI_FIRMWARE_VERSION = 2067241bc7cdf238fdeab7f5a3c22faa57dd5617
+RPI_FIRMWARE_VERSION = f9f7b94ff84947784eebb7990b80255bf0a8d94c
 RPI_FIRMWARE_SITE = $(call github,raspberrypi,firmware,$(RPI_FIRMWARE_VERSION))
 RPI_FIRMWARE_LICENSE = BSD-3c
 RPI_FIRMWARE_LICENSE_FILES = boot/LICENCE.broadcom

diff -Naur rpi-firmware.org/rpi-firmware.hash rpi-firmware/rpi-firmware.hash
--- a/package/rpi-firmware/rpi-firmware.hash	2017-03-20 11:02:15.868421552 +0000
+++ b/package/rpi-firmware/rpi-firmware.hash	2017-03-20 11:02:43.662515555 +0000
@@ -1,2 +1,3 @@
 # Locally computed
 sha256 0451e1a843bf02df81126a1625d9a15dfdf19dcb2f9e95b0613a63e59fd31ca8 rpi-firmware-2067241bc7cdf238fdeab7f5a3c22faa57dd5617.tar.gz
+sha256 8fe5ebe14cc900f67242cab0c1c9f8e29bd10369ccbdfc7d004c294a96bad975 rpi-firmware-086a8488de0f485e8879ac14ff505ff1d922eeed.tar.gz
