diff -Naur linux-headers.orig/Config.in.host linux-headers/Config.in.host
--- a/package/linux-headers/Config.in.host	2018-01-09 10:47:01.512968389 +0000
+++ b/package/linux-headers/Config.in.host	2018-01-09 10:47:51.642019834 +0000
@@ -74,6 +74,10 @@
 	bool "Linux 4.13.x kernel headers"
 	select BR2_TOOLCHAIN_HEADERS_AT_LEAST_4_13
 
+config BR2_KERNEL_HEADERS_4_14
+	bool "Linux 4.14.x kernel headers"
+	select BR2_TOOLCHAIN_HEADERS_AT_LEAST_4_14
+
 config BR2_KERNEL_HEADERS_VERSION
 	bool "Manually specified Linux version"
 
@@ -96,10 +100,15 @@
 	  This is used to hide/show some packages that have strict
 	  requirements on the version of kernel headers.
 
+config BR2_PACKAGE_HOST_LINUX_HEADERS_CUSTOM_4_14
+	bool "4.14.x"
+	select BR2_TOOLCHAIN_HEADERS_AT_LEAST_4_14
+
 config BR2_PACKAGE_HOST_LINUX_HEADERS_CUSTOM_4_13
 	bool "4.13.x"
 	select BR2_TOOLCHAIN_HEADERS_AT_LEAST_4_13
 
+
 config BR2_PACKAGE_HOST_LINUX_HEADERS_CUSTOM_4_12
 	bool "4.12.x"
 	select BR2_TOOLCHAIN_HEADERS_AT_LEAST_4_12
@@ -250,4 +259,5 @@
 	default "4.11.12"	if BR2_KERNEL_HEADERS_4_11
 	default "4.12.14"	if BR2_KERNEL_HEADERS_4_12
 	default "4.13.16"	if BR2_KERNEL_HEADERS_4_13
+	default "4.14.12"	if BR2_KERNEL_HEADERS_4_14
 	default BR2_DEFAULT_KERNEL_VERSION if BR2_KERNEL_HEADERS_VERSION
diff -Naur toolchain.orig/toolchain-common.in toolchain/toolchain-common.in
--- a/toolchain/toolchain-common.in	2018-01-09 12:02:08.809365643 +0000
+++ b/toolchain/toolchain-common.in	2018-01-09 12:00:39.620779329 +0000
@@ -253,10 +253,15 @@
 	bool
 	select BR2_TOOLCHAIN_HEADERS_AT_LEAST_4_12
 
+config BR2_TOOLCHAIN_HEADERS_AT_LEAST_4_14
+	bool
+	select BR2_TOOLCHAIN_HEADERS_AT_LEAST_4_13
+
 # This order guarantees that the highest version is set, as kconfig
 # stops affecting a value on the first matching default.
 config BR2_TOOLCHAIN_HEADERS_AT_LEAST
 	string
+	default "4.14" if BR2_TOOLCHAIN_HEADERS_AT_LEAST_4_14
 	default "4.13" if BR2_TOOLCHAIN_HEADERS_AT_LEAST_4_13
 	default "4.12" if BR2_TOOLCHAIN_HEADERS_AT_LEAST_4_12
 	default "4.11" if BR2_TOOLCHAIN_HEADERS_AT_LEAST_4_11
diff -Naur toolchain.orig/toolchain-external/toolchain-external-custom/Config.in.options toolchain/toolchain-external/toolchain-external-custom/Config.in.options
--- a/toolchain/toolchain-external/toolchain-external-custom/Config.in.options	2018-01-09 12:02:08.809365643 +0000
+++ b/toolchain/toolchain-external/toolchain-external-custom/Config.in.options	2018-01-09 12:03:01.171510529 +0000
@@ -93,6 +93,10 @@
 	    m = ( LINUX_VERSION_CODE >> 8  ) & 0xFF
 	    p = ( LINUX_VERSION_CODE >> 0  ) & 0xFF
 
+config BR2_TOOLCHAIN_EXTERNAL_HEADERS_4_14
+	bool "4.14.x"
+	select BR2_TOOLCHAIN_HEADERS_AT_LEAST_4_14
+
 config BR2_TOOLCHAIN_EXTERNAL_HEADERS_4_13
 	bool "4.13.x"
 	select BR2_TOOLCHAIN_HEADERS_AT_LEAST_4_13
